godot-core/src/builtin/rect2.rs

```diff
@@ -44,6 +44,180 @@ impl Rect2 {
         }
     }
 
+    /// Returns a rectangle with equivalent position and area, modified so that the top-left corner is the origin and `width` and `height` are positive.
+    ///
+    /// _Godot equivalent: `Rect2.abs()`_
+    #[inline]
+    pub fn abs(&self) -> Self {
+        Self {
+            position: self.position + Vector2::new(self.size.x.min(0.0), self.size.y.min(0.0)),
```

Comment:
should be possible to use `self.size.coord_min(Vector2::ZERO)`. there are also many other places you could use this that would make the code easier to read.